### PR TITLE
feat(es) Store and map containerPath to elastic search properly

### PR DESCRIPTION
**Pull request by chriscollins3456**
The new `containerPath` aspect stores a list of objects for each level in the path. This PR sets a `@Searchable` annotation to get each of the `id`s from the objects. However, we want to store this path as a single string separated by the unit-separator delimiter. So if the fieldType is a CONTAINER_PATH, aggregate this value into the string we want and store the correct mappings for this field type as well.

This is modeled after `browsePaths` so it should work very similarly is the hope.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: test-models/src/main/pegasus/com/datahub/test/TestEntityAspect.pdl [hunk 1]
```diff
@@ -8,4 +8,5 @@ typeref TestEntityAspect = union[
   TestEntityInfo,
   TestBrowsePaths,
   SearchFeatures
```
Comment (david-leifker) [inline test-models/src/main/pegasus/com/datahub/test/TestEntityAspect.pdl:10]: Missing comma?

**Review by david-leifker — APPROVED** [review]
Approving, but please fix the test before merge, likely related to the comma.

File: test-models/src/main/pegasus/com/datahub/test/TestEntityAspect.pdl (hunk 1 above)
Comment (chriscollins3456) [reply test-models/src/main/pegasus/com/datahub/test/TestEntityAspect.pdl:10]: oo nice thanks!

File: metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/MappingsBuilder.java [hunk 2]
```diff
@@ -121,7 +121,15 @@ private static Map<String, Object> getMappingsForField(@Nonnull final Searchable
...
       mappingForField.put(ANALYZER, BROWSE_PATH_HIERARCHY_ANALYZER);
       mappingForField.put(FIELDDATA, true);
-    } else if (fieldType == FieldType.URN || fieldType == FieldType.URN_PARTIAL) {
+    } else if (fieldType == FieldType.CONTAINER_PATH) {
+      mappingForField.put(TYPE, TEXT);
+      mappingForField.put(FIELDS,
+          ImmutableMap.of(LENGTH, ImmutableMap.of(
+              TYPE, TOKEN_COUNT,
+              ANALYZER, UNIT_SEPARATOR_PATTERN_ANALYZER)));
+      mappingForField.put(ANALYZER, CONTAINER_PATH_HIERARCHY_ANALYZER);
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/MappingsBuilder.java:130]: nice!!

File: metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/SettingsBuilder.java [hunk 3]
```diff
@@ -283,6 +286,13 @@ private static Map<String, Object> buildTokenizers() {
                 .put(PATTERN, "[/]")
                 .build());
 
+
+    tokenizers.put(UNIT_SEPARATOR_TOKENIZER,
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/indexbuilder/SettingsBuilder.java:290]: niceee

File: metadata-io/src/main/java/com/linkedin/metadata/search/transformer/SearchDocumentTransformer.java [hunk 4]
```diff
@@ -197,4 +208,24 @@ private Optional<JsonNode> getNodeForValue(final DataSchema.Type schemaFieldType
             : Optional.of(JsonNodeFactory.instance.textNode(fieldValue.toString()));
     }
   }
+
+  /**
+   * The containerPath aspect is a list of objects and the @Searchable annotation specifies a
+   * list of strings that we receive. However, we want to aggregate those strings and store
+   * as a single string in ElasticSearch so we can do prefix matching against it.
+   */
+  private String getContainerPathValue(@Nonnull final List<Object> fieldValues) {
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/search/transformer/SearchDocumentTransformer.java:217]: nice!

File: metadata-io/src/test/java/com/linkedin/metadata/search/transformer/SearchDocumentTransformerTest.java [hunk 5]
```diff
@@ -52,6 +53,8 @@ public void testTransform() throws IOException {
     assertEquals(browsePaths.get(1).asText(), "d/e/f");
     assertEquals(parsedJson.get("feature1").asInt(), 2);
     assertEquals(parsedJson.get("feature2").asInt(), 1);
+    JsonNode containerPath = (JsonNode) parsedJson.get("containerPath");
```
Comment (jjoyce0510) [inline metadata-io/src/test/java/com/linkedin/metadata/search/transformer/SearchDocumentTransformerTest.java:56]: awesomeee!

**Review by jjoyce0510 — APPROVED** [review]
Overall LGTM

**Pull request merged**